subsys/zigbee/osif/zb_nrf_serial.c

```diff
@@ -43,364 +20,187 @@ static serial_recv_data_cb_t rx_data_cb;
 static serial_send_data_cb_t tx_data_cb;
 static serial_send_data_cb_t tx_trx_data_cb;
 
-static uint8_t *uart_tx_buf;
-static size_t uart_tx_buf_len;
 static uint8_t uart_tx_buf_mem[CONFIG_ZIGBEE_UART_TX_BUF_LEN];
+static size_t uart_tx_buf_size;
+
+static uint8_t *uart_tx_buf;
+static uint8_t *uart_tx_buf_bak;
+static volatile size_t uart_tx_buf_offset;
+static volatile size_t uart_tx_buf_len;
+
+static uint8_t uart_rx_buf_mem[CONFIG_ZIGBEE_UART_RX_BUF_LEN];
+static struct ring_buf rx_ringbuf;
```

Comment:
This is not about the ring buffer implementation. The first commit is the result of a stabilization of the UART async-API based implementation. Since the interrupt-driven API is a bit "legacy", I wanted to keep the working "async" version in commit history, so if one day we decide that the async-API is sufficient (in details: the USB CDC class will implement it), it will be a matter of a single revert instead of going through the stabilization process again.
If one asks - yes, the first commit brings the implementation into a working, verified version and the commit bisection shall be fine with it.

Reply:
Well, it's a bit confusing, but if you say it has to be like this then I don't push further.